apps/src/client.rs

```diff
@@ -485,15 +520,79 @@ pub fn connect(
             scid_sent = true;
         }
 
+        // Performs server side migration - client kicks it off.
         if args.perform_migration &&
-            !new_path_probed &&
+            !server_preferred_address_probed &&
             scid_sent &&
             conn.available_dcids() > 0
         {
+            // Probe the transport parameter's server preferred address.
+            if let Ok(Some(preferred_address_params)) =
+                conn.server_preferred_address_params()
+            {
+                let preferred_address_params = preferred_address_params.clone();
+
+                // Manually track the new Destination Connection ID and reset
+                // token of the preferred address.
+                conn.new_destination_cid(
+                    preferred_address_params.connection_id,
+                    1,
+                    preferred_address_params.stateless_reset_token,
+                    0,
+                    &mut SmallVec::new(),
+                )
+                .unwrap();
+
+                // Probe the path. If it successfully validates then the event
+                // handler will see the `PathEvent::Validated`
+                // event occur and begin migration. The server will see
+                // `PathEvent::PeerMigrated` upon successful
+                // migration.
+                if let Some(addr_v4) = preferred_address_params.addr_v4 {
+                    info!("Performing server side connection migration - kicked off by client.");
+                    info!(
+                        "New path to be probed: {:?} -> {:?}",
+                        local_addr, addr_v4
+                    );
+                    conn.probe_path(
+                        local_addr,
+                        std::net::SocketAddr::V4(addr_v4),
+                    )
+                    .unwrap();
+                } else if let Some(addr_v6) = preferred_address_params.addr_v6 {
+                    info!("Performing server side connection migration - kicked off by client.");
+                    info!(
+                        "New path to be probed: {:?} -> {:?}",
+                        local_addr, addr_v6
+                    );
+                    conn.probe_path(
+                        local_addr,
+                        std::net::SocketAddr::V6(addr_v6),
+                    )
+                    .unwrap();
+                }
+
+                server_preferred_address_probed = true;
+            }
+        }
+
+        // Perform client side connection migration
+        if args.perform_migration &&
```

Comment:
Also consider making this its own function

Reply:
I don't quite follow, what is your reasoning to break this out into a function?

Reply:
Lucas: "Split it out to enforce soc"